tsl/src/remote/txn_resolve.c

```diff
@@ -18,14 +19,26 @@
 RemoteTxnResolution
 remote_txn_resolution(Oid foreign_server, const RemoteTxnId *transaction_id)
 {
-	if (remote_txn_is_still_in_progress(transaction_id->xid))
-		/* transaction still ongoing; don't know it's state */
+	if (remote_txn_is_still_in_progress_on_access_node(transaction_id->xid))
+		/* transaction still ongoing; don't know its state */
 		return REMOTE_TXN_RESOLUTION_UNKNOWN;
 
+	/*
+	 * If an entry exists in the "remote_txn" table and is visible then it means
+	 * that the transaction committed on the AN
```

Comment:
That is a bit confusing what is actually commit means here. Is it `PREPARE TRANSACTION`? That does not mean, that the transaction committed actually, it means that the all nodes in the cluster prepared the transaction and it is ready to be commited in the next stage, which will be `COMMIT PREPARED`.

Reply:
The comment says committed in the AN, so that means all stages including PREPARE TRANSACTION and COMMIT PREPARED have already completed on all the datanodes.